### DISCUSSION TITLE
Readonly condition if role different from status

### DISCUSSION THREAD

**Discussion by mhdaouas** [directus/directus#8230] [Archive]
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues) that already describes my problem.

### Describe the Bug

I have a workflow where I use statuses on a specific content type that have specific roles as values. I want users who are still waiting for their workflow step to get fields as readonly. I tried this condition without success:
```
{
    "status":{
       "_neq":"$CURRENT_USER.role"
    }
}
```
Is my syntax correct?

### To Reproduce

See above

### What version of Directus are you using?

9.1.4

### What version of Node.js are you using?

14.16.1

### What database are you using?

Postgresql 13

### What browser are you using?

Chrome

### What operating system are you using?

Windows

### How are you deploying Directus?

Locally on Visual Code

**Comment by benhaynes** [comment]
Try: `$CURRENT_ROLE` instead.

Also, this is not really a bug report, but more of a question on syntax... which should really be a GitHub Discussion.

**Comment by mhdaouas** [comment]
Thank you very much! You saved me! Sorry I couldn't find the right tag for the issue. Should I delete it or close it?

**Reply by azrikahar** [reply]
no worries. converted to discussion 👍

**Comment by mhdaouas** [comment]
It's still not working for me. I have created a field that needs to be readonly for all users except one so I set this rule:
`{
      "$CURRENT_ROLE":{
          "_neq":"<role primary key>"
      }
}` with a readonly option
Is there something wrong with my syntax?

**Reply by rijkvanzanten** [reply]
Those [dynamic variables](https://docs.directus.io/reference/filter-rules/#dynamic-variables) (like `$CURRENT_ROLE`) can be used as a _value_ in the filter rules setup, like

```json
// valid
{
  "role_created": {
    "_eq": "$CURRENT_ROLE"
  }
}
```

but can't be used as a _field_:

```json
// invalid
{
  "$CURRENT_ROLE": {
    "_eq": "3eddf357-c6fc-4013-834e-73759555ac84"
  }
}
```

**Reply by mhdaouas** [reply]
The problem is that I want to check the role of the user who is using the details page and checking the values of the attributes of an object.

From what I understood, the field "role_created" looks like the field which stores the role of the user who created the object but that isn't exactly my need, I want to check the role of the user that visits the detail page of the object to make specific fields read-only for specific roles.

**Reply by rijkvanzanten** [reply]
Yup! I hear ya. We normally point to / use permissions for that. If the role doesn't have create/update permission to those fields, they'll be read-only 🙂

**Reply by mhdaouas** [reply]
Use permissions => you mean in an API hook or directly from the web interface ? Is this feature (access to current user role) planned to be implemented in the future?

**Reply by rijkvanzanten** [reply]
> Use permissions => you mean in an API hook or directly from the web interface ?

In settings > roles & permissions you can configure what role has access to what. If you click on the update permissions there for your collection, you can prevent the given role from making updates to particular fields 👍🏻 

> Is this feature (access to current user role) planned to be implemented in the future?

Not sure if that's something we can easily implement from a technical perspective. The filter structure currently is "field operator value". Dynamically injecting a value is very doable, though dynamically overriding the available fields with custom values becomes way more of a technical problem. Also not sure if it's necessarily needed, seeing the end goal use case can already be achieved. I think i'd prefer looking into ways to optimize the UX so you can control role permissions from the field setup flow for example, instead of trying to reimplement the same functionality in a different way 🙂

**Reply by mhdaouas** [reply]
The problem is that the user will believe that he can modify the values but in reality he cannot and realizes that only after clicking on the update button. I think the only annoying inconvenient is that it can waste time for the user, especially for big forms.

**Reply by rijkvanzanten** [reply]
That's not true. If you don't have permission to update a field, it will render as readonly 👍🏻

**Reply by mhdaouas** [reply]
Ah ok, I haven't tried that much the feature you talked about (user permissions), will let you know soon if it works for me. Thanks for your continuous support.

**Reply by mhdaouas** [reply]
It works like a charm! Thanks!
